src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/ActivitiesMigrationService.kt

```diff
@@ -108,6 +108,28 @@ class ActivitiesMigrationService(
     migrationHistoryService.updateFilter(migrationId, filter)
   }
 
+  suspend fun moveActivityStartDates(migrationId: String, newActivityStartDate: LocalDate): List<String> {
+    val activityCount = activitiesMappingService.getMigrationCount(migrationId)
+    if (activityCount == 0L) throw NotFoundException("No migrations found for $migrationId")
+
+    val allActivityIds = activitiesMappingService.getActivityMigrationDetails(migrationId, activityCount).content
+      .map { it.nomisCourseActivityId }
+
+    val filter = migrationHistoryService.get(migrationId)
+      .let { objectMapper.readValue(it.filter, ActivitiesMigrationFilter::class.java) }
+
+    return try {
+      nomisApiService.endActivities(allActivityIds, newActivityStartDate.minusDays(1))
+        .also { filter.nomisActivityEndDate = newActivityStartDate.minusDays(1) }
+      activitiesApiService.moveActivityStartDates(filter.prisonId, newActivityStartDate)
+        .also { filter.activityStartDate = newActivityStartDate }
+    } catch (e: Exception) {
+      throw MoveActivityStartDatesException(e)
+    } finally {
+      migrationHistoryService.updateFilter(migrationId, filter)
+    }
+  }
```

Comment:
This is the crux of the change - call NOMIS and DPS to adjust activity end dates / start dates, and update the migration filter so this is visible to users in the UI.